src/trade.js

```diff
@@ -152,7 +161,8 @@ class Trade {
     var reservation = quote.delegate.reserveReceiveAddress();
 
     var processTrade = function (res) {
-      var trade = new quote._TradeClass(res, quote.api, quote.delegate);
+      var trade = new quote._TradeClass(null, quote.api, quote.delegate);
+      trade.setFromAPI(res);
```

Comment:
`setFromAPI` doesn't seem to take any arguments. And why pass `null` to the constructor just to call a method with the trade object right after? If initializing from metadata / api / elsewhere are really that different, I would rather use static methods `fromMetadata`, `fromAPI`, etc, that return Trade instances.